packages/wallet-apis/src/provider.ts

```diff
@@ -0,0 +1,207 @@
+) => {
+  // TODO(v5): implement any other supported events: https://eips.ethereum.org/EIPS/eip-1193#events
+  const eventEmitter = new EventEmitter();
+
+  // TODO(v5): revisit if we actually want to auto-connect the provider like this.
+  (async () => {
+    const account = await client.requestAccount();
+    client.account = {
+      type: "json-rpc",
+      address: account.address,
+    };
+    eventEmitter.emit("connect", { chainId: client.chain.id });
+  })();
+
+  const request = (async ({ method, params }) => {
+    switch (method) {
+      case "eth_chainId": {
+        return handler<"eth_chainId">(async () => {
+          if (!client.chain.id) {
+            throw new ChainNotFoundError();
+          }
+          return toHex(client.chain.id);
+        })(params);
+      }
+
+      case "eth_accounts": {
+        return handler<"eth_accounts">(async () => {
+          if (!client.account) {
+            throw new AccountNotFoundError();
+          }
+          return [client.account.address];
+        })(params);
+      }
+
+      case "personal_sign": {
+        return handler<"personal_sign">(async ([data, address]) => {
+          if (!client.account) {
+            throw new AccountNotFoundError();
+          }
+          if (
+            address?.toLowerCase() !== client.account.address?.toLowerCase()
+          ) {
+            throw new BaseError(
+              "Cannot sign for an address other than the current account.",
+            );
+          }
+          return await client.signMessage({
+            message: {
+              raw: data,
+            },
+            account: client.account.address,
+          });
+        })(params);
+      }
+
+      case "eth_signTypedData_v4": {
+        return handler<"eth_signTypedData_v4">(async ([address, tdJson]) => {
+          if (!client.account) {
+            throw new AccountNotFoundError();
+          }
+          if (address?.toLowerCase() !== client.account.address.toLowerCase()) {
+            throw new Error(
+              "Cannot sign for an address other than the current account.",
+            );
+          }
+          return await client.signTypedData({
+            account: client.account.address,
+            ...(JSON.parse(tdJson) as TypedDataDefinition),
+          });
+        })(params);
+      }
+
+      // eslint-disable-next-line no-fallthrough
+      case "wallet_sendTransaction":
+      case "eth_sendTransaction": {
+        return handler<"eth_sendTransaction">(async ([tx]) => {
+          if (!client.account) {
+            throw new AccountNotFoundError();
+          }
+          if (!client.chain) {
+            throw new ChainNotFoundError();
+          }
+          const { to, data, value } = tx;
+          if (!to) {
+            throw new BaseError("'to' is required.");
+          }
+          const result = await client.sendCalls({
+            calls: [{ to, data, value }],
+            from: client.account.address,
```

Comment:
Do you think we should check the from that is passed to the txn like this?  I think I've seen this in other providers.  It might be confusing if someone can just pass a random from value and things still work.

```
if (tx.from && tx.from.toLowerCase() !== client.account.address.toLowerCase()) {
  throw new Error()
}
```